### ISSUE TITLE
Fails to resubscribe after `ECONNRESET`

### ISSUE THREAD

**Issue by mastermatt** [kibae/pg-logical-replication#20]
I'm posting this more to see if anyone else has seen this issue. 

I've been having a hard time replicating the the conditions in a test scenario, but in deployed environments a `LogicalReplicationService` instance hangs on the `subscribe` call after erroring from a socket reset (`ECONNRESET`).

This lib seems to be doing everything correctly in the `subscribe` method to ensure the client and connection instance states are clean before attempting to reconnect, but I just never get any more events after that point. 

@kibae have you come across this by chance? 

NodeJS: 16.19
OS: Alpine Linux: 3.17
PostgreSQL: 14.4

**Comment by Zehelein** [comment]
I created a transactional outbox library based on this library. I added a few integration tests to shut down the PG server to mimick some outages. It produced errors from which this library did not recover most of the time.

You can have a look here on the tests: https://github.com/Zehelein/pg-transactional-outbox/blob/master/__tests__/src/outbox-inbox-resilience.spec.ts

I saw some issues arise if there was a PG based error as it was triggering the error on the logical replication itself as well as on the subscription. I solved it basically the brute-force way to recreate the logical replication service including the connection: https://github.com/Zehelein/pg-transactional-outbox/blob/master/lib/src/local-replication-service.ts

So in summary: I was also not able to figure out why the PG connection could not re-establish itself after a hard error.

**Comment by mastermatt** [comment]
I believe I've gotten to the root cause of the issue and have created an issue against `pg` (linked above).

**Comment by mastermatt** [comment]
This _should_ be fixed with the latest release of pg 8.10.0. 
I'll try to find some time to verify in my code base.
